src/session/content/item_row.rs

```diff
@@ -94,16 +96,311 @@ impl ItemRow {
                 match item.type_() {
                     ItemType::Message(message) => {
                         let content = message.content().0;
-                        let message = message.to_owned().upcast();
+                        let sender_name = match message.sender().to_owned() {
+                            MessageSender::User(data) => super::item_row::sender_name(&data),
+                            MessageSender::Chat(data) => data.title(),
+                        };
+                        let message_obj = message.to_owned().upcast();
```

Comment:
The `message_obj` is only needed in 3 match arms. Could we upgrade them in the place where we require it?